corehq/apps/data_dictionary/templates/data_dictionary/base.html

```diff
@@ -4,9 +4,7 @@
 
 {% block title %}{% trans "Data Dictionary" %}{% endblock %}\
 
-{% block js %}
-<script src="{% static 'data_dictionary/js/data_dictionary.js' %}"></script>
-{% endblock %}
+{% block requirejs_main %}data_dictionary/js/data_dictionary{% endblock requirejs_main %}
```

Comment:
Could this be changed to `{% set_requirejs_main request "data_dictionary/js/data_dictionary" %}`, and then remove some code from [hqwebapp/base.html](https://github.com/dimagi/commcare-hq/pull/17824/commits/c34703accfb1f66547ada7faa10bb7fc52e61a44#diff-1ecb20ffccb745a5c0fc279837215a25) (see comment).

Nitpick: if you do this, what do you think of renaming the tag to simply `requirejs_main`?

Reply:
I started with code similar to that, but then the `{% set_requirejs_main ... %}` has to be inside of some other block, while still being guaranteed to run before any `if request.requirejs_main` conditions happen. Making requirejs config its own block guarantees `request.requirejs_main` gets set early enough, which seemed more robust. Thoughts?

Reply:
I started working on an implementation last night. I'll try to get it into this PR. If I can't get it done in time, I'll do it as a separate PR after this has been merged.

Reply:
#17903